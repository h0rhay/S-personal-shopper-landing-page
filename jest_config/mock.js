export const mockTheme = {
  themeType: 'dark',
  breakpoints: {
    small: '(max-width: 812px)',
    medium: '(min-width: 813px)',
    large: '(min-width: 960px)',
    deviceSwitchSm: '(max-width: 896px)',
    deviceSwitchLg: '(min-width: 897px)',
  },
  typography: {
    mobile: {
      headerOne: {
        fontSize: '3rem',
      },
      headerTwo: {
        fontSize: '1.75rem',
      },
      displayHeading: {
        fontSize: 'calc(1.5rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
    desktop: {
      headerOne: {
        fontSize: '4rem',
      },
      headerTwo: {
        fontSize: '2rem',
      },
      displayHeading: {
        fontSize: 'calc(4rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
  },
  text: '#ffffff',
  primary: '#212121',
  primaryAccent: '#2E2E2E',
  accent: '#ffe256',
  shadow: 'rgba(0, 0, 0, 0.4)',
};
