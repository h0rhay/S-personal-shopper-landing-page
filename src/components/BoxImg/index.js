import PropTypes from 'prop-types';
import React from 'react';
import styled, { css } from 'styled-components';

const BoxImgStyle = styled.div`
  position: relative;
  &:before {
    z-index: -1;
    content: '';
    position: absolute;
    width: 100%;
    height: 100%;
    ${(props) =>
      !props.dir || props.dir === 'right'
        ? css`
            top: -1rem;
            right: -1rem;
          `
        : css`
            bottom: -1rem;
            left: -1rem;
          `}
    ${(props) =>
      props.dir &&
      props.dir === 'shopper' &&
      css`
        top: -2rem;
        left: -2rem;
      `}
    outline: 1px solid ${({ theme }) => theme.accent};
  }
`;

const BoxImg = ({ children, dir, ...delegated }) => (
  <BoxImgStyle {...delegated} dir={dir}>
    {children}
  </BoxImgStyle>
);

BoxImg.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.string,
  dir: PropTypes.string,
};

export default BoxImg;
