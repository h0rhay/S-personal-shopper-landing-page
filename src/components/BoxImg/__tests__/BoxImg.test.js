import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { mockTheme } from '../../../../jest_config/mock';
import BoxImg from '../index';

const props = {
  dir: 'right',
};

const renderComponent = ({ theme }) =>
  render(
    <ThemeProvider theme={theme}>
      <BoxImg {...props} />
    </ThemeProvider>
  );

describe('BoxImg component', () => {
  it('should match screenshot', () => {
    const { container } = renderComponent({ theme: mockTheme, props });
    expect(container).toMatchSnapshot();
  });
});
