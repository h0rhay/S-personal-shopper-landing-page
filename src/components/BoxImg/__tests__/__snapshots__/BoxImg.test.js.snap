// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxImg component should match screenshot 1`] = `
<div>
  <div
    class="sc-bdvvtL ldnvIA"
    dir="right"
  />
</div>
`;
